doc/design/executor.md

```diff
@@ -0,0 +1,32 @@
+# Executor Desgin Doc
+
+## Overview
+
+`Executor` evaluates a `ProgramDesc`. Essentially, it instantializes Variables and Operators, then run all the operators
+
+```c++
+void Executor::Run(const ProgramDesc& pdesc, Scope* scope, int block_id) {
+  auto& block = pdesc.blocks(block_id);
+  auto& device = device_contexts_[0];
+
+  // Instantiate all the vars in the global scope
+  for (auto& var : block.vars()) {
+    scope->NewVar(var.name());
```

Comment:
A `pdesc` may run multiple times, do we need to instantiate the vars every time?

Reply:
if the `var.name()` already exists, it won't be re-created.

Reply:
Not related to this PR, I think `NewVar` should return an error if no new var is created, otherwise it should not be call `NewVar`.